tests/components/mqtt/test_config_flow.py

```diff
@@ -183,15 +183,14 @@ async def test_user_connection_works(
     assert result["type"] == "form"
 
     result = await hass.config_entries.flow.async_configure(
-        result["flow_id"], {"broker": "127.0.0.1", "advanced_options": False}
```

Comment:
"advanced_options" was filtered out, that was why tests pass, it should not be a setting in the entry, and in reallive this setting was never added to the MQTT config entry data.